worlds/stardew_valley/logic/skill_logic.py

```diff
@@ -197,13 +188,21 @@ def can_forage_quality(self, quality: str) -> StardewRule:
             return self.has_level(Skill.foraging, 9)
         return False_()
 
-    @cached_property
-    def can_earn_mastery_experience(self) -> StardewRule:
-        if self.options.skill_progression != options.SkillProgression.option_progressive_with_masteries:
-            return self.has_all_skills_maxed() & self.logic.time.has_lived_max_months
-        return self.logic.time.has_lived_max_months
+    def can_earn_mastery(self, skill: str) -> StardewRule:
+        # Checking for level 11, so it includes having level 10 and being able to earn xp.
+        return (self.logic.skill.can_earn_level(skill, 11) &
+                self.logic.time.has_lived_max_months &

```

Comment:
I think this max months rule will probably hurt the design in the long run, by forcing all masteries to be late regardless of which ones are accessible earlier. I'd rather a flat number of months, or even no months rule at all, and assuming that "can earn level 11" implies a scaled number of months that is reasonably balanced (bigger than level 10, but smaller than levels 1-10)

Reply:
Sounds right. I changed that to only check for months when checking the vanilla rule, so now it's entirely based on having the level and no longer require time if skills are randomized.